W-17456680 Adding Salesforce connector example (#623)
mulesoft/docs-code-builder: modules/ROOT/pages/_partials/acb-component-info.adoc

//
// tag::add-components[]

> The following examples illustrate basic configurations for adding components to your project from the canvas, and the configuration XML. The examples assume you are beginning with an empty integration project.

Reviewer: ```suggestion
These examples illustrate basic configurations for adding components to your project from the canvas, and the configuration XML. The examples assume you are beginning with an empty integration project.
```

+
.. From the canvas, click the image:icon-plus.png["",15,15] (*Add component*) icon after *Set payload* (or *Transform Message*).
.. In the *Add Component* panel, click *Connectors*, select *Salesforce*, and select *Create*.
.. In the configuration panel for *Create*, set:
+
--
* *Connection config*: The global Salesforce configuration name (for example, `Salesforce_Config`).
* *Type*: `Account`.
* *Records*: A DataWeave expression that resolves to the records to create.
+
If the incoming payload uses different property names than Salesforce expects, map those properties in *Records*, as shown in the configuration XML example that follows.
If the payload is already an array of Java objects with Salesforce field names (`output application/java`), you can use `#[payload]`.
--
+
[TIP]
====
If *Type* stays empty until you establish a working Salesforce connection, set `type="Account"` on `<salesforce:create/>` in the configuration XML.
For more information about connector fields for this operation, see xref:salesforce-connector::index.adoc[].
====
+
[NOTE]
====
Picklist fields such as *Industry* must use values that exist in your Salesforce organization.
====
+

> The following configuration XML listens on `/accounts`, builds a JSON payload, maps it to Account fields inside `<salesforce:records/>`, and calls *Create*.

Reviewer: ```suggestion
This configuration XML listens on `/accounts`, builds a JSON payload, maps it to Account fields inside `<salesforce:records/>`, and calls *Create*.
```
